openedx/core/djangoapps/catalog/utils.py

```diff
@@ -98,9 +104,14 @@ def get_programs(site, uuid=None):
             logger.warning(missing_details_msg_tpl.format(uuid=uuid))
 
         return program
-    uuids = cache.get(SITE_PROGRAM_UUIDS_CACHE_KEY_TPL.format(domain=site.domain), [])
-    if not uuids:
-        logger.warning(u'Failed to get program UUIDs from the cache for site {}.'.format(site.domain))
+    elif course:
+        uuids = cache.get(COURSE_PROGRAMS_CACHE_KEY_TPL.format(course_run_id=course))
```

Comment:
I'd add a default to the `get` call like the rest of them. Otherwise the function sometimes returns `None` and sometimes `[]` on cache misses.